app/server/appsmith-server/src/main/java/com/appsmith/server/services/ConsolidatedAPIServiceImpl.java

```
@@ -61,6 +61,7 @@ public class ConsolidatedAPIServiceImpl implements ConsolidatedAPIService {
             "release_server_dsl_migrations_enabled";
     public static final int INTERNAL_SERVER_ERROR_STATUS = AppsmithError.INTERNAL_SERVER_ERROR.getHttpErrorCode();
     public static final String INTERNAL_SERVER_ERROR_CODE = AppsmithError.INTERNAL_SERVER_ERROR.getAppErrorCode();
+    public static final String EMPTY_WORKSPACE_ID_ON_ERROR = "";
```

@sumitsum For my knowledge, why did we have to introduce this error? In what scenario would the workspace ID be empty?

---

Hi @mohanarpit this is a value that is used to identify whether the code flow to fetch workspace id has failed (for any reason). This value is not supplied to the downstream APIs. It is instead used as an indicator of failure to fetch workspace id and hence not supplying the workspace id to downstream APIs. Example:
```
MultiValueMap<String, String> params = new LinkedMultiValueMap<>();
if (!EMPTY_WORKSPACE_ID_ON_ERROR.equals(workspaceId)) {
    params.add(WORKSPACE_ID, workspaceId);
}
return datasourceService.getAllWithStorages(params).collectList();
```

The idea here is that the downstream Mono should return a validation error instead of the error that is returned from workspace id Mono. ([Slack thread](https://theappsmith.slack.com/archives/C024GUDM0LT/p1704355122219259?thread_ts=1703074675.245879&cid=C024GUDM0LT))